core/src/main/java/hudson/model/Slave.java

```diff
@@ -162,6 +163,16 @@ public Slave(String name, String nodeDescription, String remoteFS, int numExecut
     	this(name, nodeDescription, remoteFS, numExecutors, mode, labelString, launcher, retentionStrategy, new ArrayList());
     }
 
+    public Slave(@Nonnull String name, String remoteFS, ComputerLauncher launcher) throws FormException, IOException {
```

Comment:
All the other constructors on this abstract class is public, so I think there's a valid argument to be made both ways. I think it's fine as is.